frontend/src/diagram/DropArea.tsx

```diff
@@ -0,0 +1,172 @@
+
+const toolType = PropTypes.shape({
+  id: PropTypes.string.isRequired,
+  label: PropTypes.string.isRequired,
+  imageURL: PropTypes.string,
+});
+const propTypes = {
+  representationId: PropTypes.string.isRequired,
+  toolSections: PropTypes.arrayOf(
+    PropTypes.shape({
+      label: PropTypes.string.isRequired,
+      tools: PropTypes.arrayOf(toolType).isRequired,
+      defaultTool: toolType.isRequired,
+    })
+  ).isRequired,
+  invokeHover: PropTypes.func.isRequired,
+  children: PropTypes.node.isRequired,
+};
+
+export const DropArea = ({ representationId, toolSections, invokeHover, children }) => {
+  const { id } = useProject();
+
+  const [invokeDropTool] = useMutation(invokeDropToolOnDiagramMutation);
+
+  const dropElement = useCallback(
+    (tool: any, objectId: string, diagramElementId?: string) => {
+      const { id: toolId } = tool;
+      const input = {
+        projectId: id,
+        representationId,
+        objectId,
+        toolId,
+      };
+      if (diagramElementId) {
+        input['diagramElementId'] = diagramElementId;
+      }
+
+      const variables = {
+        input,
+      };
+      invokeDropTool({ variables });
+    },
+    [id, representationId, invokeDropTool]
+  );
+
+  const searchId = useCallback((dom) => {
+    if (dom.hasAttribute(DATA_DIAGRAM_ID)) {
+      return dom.getAttribute(DATA_DIAGRAM_ID);
+    } else if (dom.hasAttribute(DATA_NODE_ID)) {
+      return dom.getAttribute(DATA_NODE_ID);
+    } else if (dom.parentElement) {
+      const parentDom = dom.parentElement;
+      if (parentDom.namespaceURI === SVG_NAMESPACE) {
+        return searchId(parentDom);
+      }
+    }
+    return null;
+  }, []);
+
+  const searchDescriptionId = useCallback((dom) => {
+    if (dom.hasAttribute(DATA_DESCRIPTION_ID)) {
+      return dom.getAttribute(DATA_DESCRIPTION_ID);
+    } else if (dom.parentElement) {
+      const parentDom = dom.parentElement;
+      if (parentDom.namespaceURI === SVG_NAMESPACE) {
+        return searchDescriptionId(parentDom);
+      }
+    }
+    return null;
+  }, []);
+
+  const searchDropTool = useCallback(
+    (targetId, targetDescriptionId) => {
+      let dropTool;
+      for (let toolSection of toolSections) {
+        const dropTools = toolSection.tools.filter((tool) =>
+          canDrop(tool, targetId, targetDescriptionId, representationId)
+        );
+        if (dropTools && dropTools.length > 0) {
+          dropTool = dropTools[0];
+          break;
+        }
+      }
+      return dropTool;
+    },
+    [representationId, toolSections]
+  );
+
+  const handleDragOver = useCallback(
+    (e) => {
+      e.preventDefault();
+      const id = searchId(e.target);
+      invokeHover(id, true);
+      const descriptionId = searchDescriptionId(e.target);
+      const tool = searchDropTool(id, descriptionId);
+      if (tool) {
+        e.dataTransfer.dropEffect = 'link';
+      }
+    },
+    [invokeHover, searchId, searchDescriptionId, searchDropTool]
+  );
```

Comment:
If only one tool can be used to drop an object on the diagram, why does the frontend does need to:
- receive all the drop tools available
- filter and find the proper drop tool

Could you add an ADR to describe how you are thinking about this feature.

Reply:
The purpose is to handle several drop tool in the following PR but I can simplify this code for now, in deed.(Ok for the ADR)

Reply:
After a light refactoring, it appears that we could split in several dropTool according that each dedicated dropTool can be focused on a subSet of targets.